sqlglot/tokens.py

```diff
@@ -1041,12 +1041,6 @@ def _advance(self, i: int = 1, alnum: bool = False) -> None:
     def _text(self) -> str:
         return self.sql[self._start : self._current]
 
-    def peek(self, i: int = 0) -> str:
```

Comment:
is this not used?

Reply:
Yep, after the `decimal` refactor it was dead code, see for example the warning in the gh action deployment [job](https://github.com/tobymao/sqlglot/actions/runs/8088464225/job/22102560924):

```rust
warning: method `peek` is never used
   --> src/tokenizer.rs:181:8
    |
84  | impl<'a> TokenizerState<'a> {
    | --------------------------- method in this implementation
...
181 |     fn peek(&self, i: usize) -> Result<char, TokenizerError> {
    |        ^^^^
    |
    = note: `#[warn(dead_code)]` on by default

warning: `sqlglotrs` (lib) generated 1 warning
```